packages/proto-rpc/src/ClientTransport.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    ClientStreamingCall,
+    ServerStreamingCall,
+    DuplexStreamingCall,
+    Deferred,
+    RpcTransport,
+    MethodInfo,
+    RpcError,
+    RpcOptions,
+    RpcMetadata,
+    RpcStatus,
+    UnaryCall,
+    mergeRpcOptions
+} from '@protobuf-ts/runtime-rpc'
+import { v4 } from 'uuid'
+import { RpcMessage } from './proto/ProtoRpc'
+import EventEmitter = require('events')
+import { Logger } from './Logger'
+
+export enum Event {
+    RPC_REQUEST = 'rpcRequest'
+}
+
+export interface ClientTransport {
+    on(event: Event.RPC_REQUEST, listener: (results: ResultParts, rpcMessage: RpcMessage, options: ProtoRpcOptions) => void): this
+}
+
+export interface ResultParts {
+    header: Deferred<RpcMetadata>,
+    message: Deferred<object>,
+    status: Deferred<RpcStatus>,
+    trailer: Deferred<RpcMetadata>,
```

Comment:
Is `trailer` a protobuf term, or our own term? Is it a kind of a footer?

Reply:
It is a footer used in HTTP3 / gRPC.

Reply:
Why the name is `trailer` and not `footer`?

Reply:
Naming comes from the grpc protocol.